tests/embedding/test_appdata.py

```diff
@@ -0,0 +1,109 @@
+
+
+@pytest.mark.embedding_scripts
+@pytest.mark.python_env
+def test_private_appdata(pytestconfig, run_subprocess, rootdir):
+    """Test embedded instance does not save ShowTriad using a test-scoped Python environment."""
+
+    version = pytestconfig.getoption("ansys_version")
+    embedded_py = os.path.join(rootdir, "tests", "scripts", "run_embedded_app.py")
+
+    run_subprocess([sys.executable, embedded_py, version, "True", "Set"])
+    process, stdout, stderr = run_subprocess([sys.executable, embedded_py, version, "True", "Run"])
+    stdout = stdout.decode()
+    assert "ShowTriad value is True" in stdout
+
+
+@pytest.mark.embedding_scripts
+@pytest.mark.python_env
+def test_normal_appdata(pytestconfig, run_subprocess, rootdir):
+    """Test embedded instance saves ShowTriad value using a test-scoped Python environment."""
+    version = pytestconfig.getoption("ansys_version")
+
+    embedded_py = os.path.join(rootdir, "tests", "scripts", "run_embedded_app.py")
+
+    run_subprocess([sys.executable, embedded_py, version, "False", "Set"])
+    process, stdout, stderr = run_subprocess([sys.executable, embedded_py, version, "False", "Run"])
+    run_subprocess([sys.executable, embedded_py, version, "False", "Reset"])
+
+    stdout = stdout.decode()
+    # Assert ShowTriad was set to False for regular embedded session
+    assert "ShowTriad value is False" in stdout
+
+
+@pytest.mark.embedding
+def test_uniqueprofile_creation():
+
+    folder_to_check = os.path.join("AppData", "Local", "Ansys") if os.name == "nt" else ".mw"
+
+    # Create private app data without copying profiles
+    private_data2 = UniqueUserProfile(profile_name="test1", copy_profile=False)
+    assert not os.path.exists(os.path.join(private_data2.location, folder_to_check))
+
+    # Check if location is same with same profile name
+    private_data1 = UniqueUserProfile(profile_name="test1")
+    assert private_data1.location == private_data2.location
+
+    # check if folder exists after copying profiles
+    assert os.path.exists(os.path.join(private_data1.location, folder_to_check))
+
+    # Create new profile
+    private_data3 = UniqueUserProfile(profile_name="test2")
+    assert private_data2.location != private_data3.location
+
+
+@pytest.mark.embedding
+def test_uniqueprofile_env():
+    profile = UniqueUserProfile("test_env")
+    env = {}
+    platforms = ["win32", "linux"]
+
+    for platform in platforms:
+        with mock.patch.object(sys, "platform", platform):
+            profile.update_environment(env)
+
+    if platform == "win32":
+        env["USERPROFILE"] = profile.location
+        env["APPDATA"] = os.path.join(profile.location, "AppData/Roaming")
+        env["LOCALAPPDATA"] = os.path.join(profile.location, "AppData/Local")
+        env["TMP"] = os.path.join(profile.location, "AppData/Local/Temp")
+        env["TEMP"] = os.path.join(profile.location, "AppData/Local/Temp")
+    else:
+        env["HOME"] = profile.location
+
```

Comment:
Is this missing "assert" statements to check if the environment was updated correctly?

Reply:
yes. Was there any assert missing ?

Reply:
I don't see any asserts in the test_uniqueprofile_env() function. Did you want to test the env variables like this?

Should it be `assert os.environ.get("USERPROFILE") == profile.location` instead of `env["USERPROFILE"] = profile.location` for example?

Reply:
I am passing `env` instead of passing `os.env` just to test `profile.update_environment` is working or not which we are using similar in `App()`. We can do with os.env also but I did not find its necessary.